File: SearchInsertPosition/SearchInsert.py
from typing import List

def searchInsert(nums: List[int], target: int) -> int:
        start = 0
        end = len(nums) - 1
        
        while start <= end:
            mid = start + (end - start) // 2
            if nums[mid] == target:
                return mid
            elif nums[mid] > target:
                end = mid - 1
            else:
                start = mid + 1
                
        return start


nums = list(map(int, input("Enter numbers seperated by space: ").split()))
target = int(input("enter the target number:"))

print("Index value:",  searchInsert(nums,target))

File: SearchInsertPosition/SearchInsert.java
package SearchInsertPosition;

import java.util.Scanner;

public class SearchInsert {

    public static int searchInsert(int[] nums, int target) {
        int start = 0;
        int end = nums.length - 1;

        while(start <= end) {
            int mid = start + (end - start) / 2;
            if (nums[mid] == target) {
                return mid;
            } else if(nums[mid] > target) {
                end = mid - 1;
            } else {
                start = mid + 1;
            }
        }

        return start;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter numbers separated by space:");
        String[] input = sc.nextLine().split(" ");
        int[] nums = new int[input.length];

        for (int i = 0; i < input.length; i++) {
            nums[i] = Integer.parseInt(input[i]);
        }

        System.out.println("Enter target:");
        int target = sc.nextInt();

        int result = searchInsert(nums, target);
        System.out.println("Index value: " + result);

        sc.close();
    }
}
